src/components/Canvas.vue

```diff
@@ -335,22 +395,80 @@ export default defineComponent({
       const gridder = (pos: Pos): Pos => {
         const f = (n: number) => (g == 0 ? n : Math.round((n + g / 2) / g) * g);
         return {
-          x: f(pos.x),
-          y: f(pos.y),
+          x: Math.round(f(pos.x)),
+          y: Math.round(f(pos.y)),
         };
       };
       const limiter = (pos: Pos): Pos => {
         const f = (can: number, n: number, offset: number, cur: number) =>
-          Math.max(0, Math.min(can - g - 1, n - offset + cur / 2 - 3));
+          Math.max(0, Math.min(can - g - 1, n - offset + cur / 2));
         return {
           x: f(canw, pos.x, offsetX.value, curw),
           y: f(canh, pos.y, offsetY.value, curh),
         };
       };
+      const magnification =
+        currentTool.value === Tools.ZOOM
+          ? Math.abs(pivotPos.value.y + offy - evt.pageY) /
```

Comment:
Is this safa for divide by zero ?

Reply:
Okay, I'll put some guard before. Thanks again !!